### ISSUE TITLE
Object variants: allow to be copied/cloned

### ISSUE THREAD

**Issue by dolmit-tanel-paaro** [pimcore/pimcore#8543]
# Feature Request
Allow object variants to be duplicatable.  
## Feature description
Our client has many products and product variants. Often the content managers need to create new product variants that differ very little from the existing variants. It would be a lot more efficient for them to work if the variants were easily duplicatable. Variants generally only differ by 1-2 attributes (e.g. MPN & size).

For unknown reason the copy option has been excluded from the context menu for variants in the [tree.js](https://github.com/pimcore/pimcore/blob/master/bundles/AdminBundle/Resources/public/js/pimcore/object/tree.js#L513-L530):
```
if (!isVariant) {
    if (record.data.id != 1 && record.data.permissions.view && perspectiveCfg.inTreeContextMenu("object.copy")) {
        menu.add(new Ext.menu.Item({
            text: t('copy'),
            iconCls: "pimcore_icon_copy",
            handler: this.copy.bind(this, tree, record)
        }));
    }

    //cut
    if (record.data.id != 1 && !record.data.locked && record.data.permissions.rename && perspectiveCfg.inTreeContextMenu("object.cut")) {
        menu.add(new Ext.menu.Item({
            text: t('cut'),
            iconCls: "pimcore_icon_cut",
            handler: this.cut.bind(this, tree, record)
        }));
    }
}
```
  
I would like an explanation on why this has been done.

**Comment by dolmit-tanel-paaro** [comment]
We decided to take the risk & allow the copy functionality for the variants. We created a patch that alters the core logic:
```
if (record.data.id != 1 && record.data.permissions.view && perspectiveCfg.inTreeContextMenu("object.copy")) {
    menu.add(new Ext.menu.Item({
        text: t('copy'),
        iconCls: "pimcore_icon_copy",
        handler: this.copy.bind(this, tree, record)
    }));
}
if (!isVariant) {
    //cut
    if (record.data.id != 1 && !record.data.locked && record.data.permissions.rename && perspectiveCfg.inTreeContextMenu("object.cut")) {
        menu.add(new Ext.menu.Item({
            text: t('cut'),
            iconCls: "pimcore_icon_cut",
            handler: this.cut.bind(this, tree, record)
        }));
    }
}
```

We can only hope that this does not create any new issues. Sadly we did not find any information or were able to figure out if this modification can cause any negative side effects.

**Comment by fashxp** [comment]
Dear community, it's been a while since that issue was updated the last time, so we decided to close it for now. If you're still interested in contributing, feel free to add comments here or contact us before creating a pull request to discuss next steps.

Thanks for your understanding.

**Comment by dolmit-tanel-paaro** [comment]
> Dear community, it's been a while since that issue was updated the last time, so we decided to close it for now. If you're still interested in contributing, feel free to add comments here or contact us before creating a pull request to discuss next steps.
> 
> Thanks for your understanding.

Hello,

Can you please provide an explanation on why copy feature has been removed for object variants?

**Comment by ARJUN-K-P** [comment]
hello @fashxp , 
      We even came up with the same kind of issue.  Is that okay, I am updating the code and creating the pull request?
Right now we are in PIMCore 10.6.9. We are also working on upgrading the code for version 11. So we need to fix it for both 10.6.9 and 11.

**Comment by fashxp** [comment]
Yes please
